command-line-local-shortener/main.go

```diff
@@ -35,47 +34,28 @@ func isUrl(str string) bool {
 	return u.Scheme != "" && u.Host != ""
 }
 
-func dataProccess() {
-	sc := bufio.NewScanner(os.Stdin)
+func dataProccess(inputURL string) string {
 	idLength := 3
-	for {
-
-		sc.Scan()
-		input := sc.Text()
-		fmt.Println("")
-
-		if input == "s" || input == "S" {
-			fmt.Println("Please paste your url: ")
-			sc.Scan()
-			inputURL := sc.Text()
-			if isUrl(inputURL) {
-				for {
-					currID := genID(idLength)
-					if _, exists := dataArray[currID]; !exists {
-						dataArray[currID] = inputURL
-						result, err := json.Marshal(dataArray)
-						if err != nil {
-							logrus.Errorf("error while marshalling json: %v", err)
-							os.Exit(1)
-						}
-						ioutil.WriteFile("urlmap.json", result, 0644)
-
-						for _, v := range dataArray {
-							fmt.Println(v)
-						}
-
-						break
-					} else {
-						idLength += 1
-					}
+	if isUrl(inputURL) {
+		for {
+			currID := genID(idLength)
+			if _, exists := dataArray[currID]; !exists {
+				dataArray[currID] = inputURL
+				result, err := json.Marshal(dataArray)
+				if err != nil {
+					logrus.Errorf("error while marshalling json: %v", err)
+					os.Exit(1)
```

Comment:
instead of `os.Exit` here... we want the webserver to return an error. Make this a `return "", fmt.Errorf("error marshalling json for URL map: %v", err)`